src/main/java/com/ocient/jdbc/XGConnection.java

```diff
@@ -406,7 +423,7 @@ public OAuthFlow getOAuthFlow(Properties properties, OAuthFlow defaultValue) thr
 		}
 
 		public HandshakeType getHandshakeType(Properties properties, HandshakeType defaultValue) throws SQLException {
```

Comment:
This was a poor design choice on my part... You shouldn't be forced to provide a default value here, or in any of the other getter methods. This might be a better approach for the get-or-default pattern.
```suggestion
		public <E> Optional<HandshakeType> getHandshakeType(Properties properties) throws SQLException {
```
or
```
                 public OptionalInt getInt(final Properties properties) {
```
That way, the caller can do
```
final int maxRows = ConnectionProperty.MAX_ROWS.getInt(properties).orElse(0);
if((proposedMaxRows < 1) && (proposedMaxRows != -1)){
  // throw
}
```
Thoughts?